judo-tatami-jsl-jsl2ui/src/test/java/hu/blackbelt/judo/tatami/jsl/jsl2ui/application/JslModel2UiDataTest.java

```diff
@@ -369,6 +512,171 @@ void testRelations() throws Exception {
         assertEquals(1, apps.size());
 
         Application app1 = apps.get(0);
+
+        assertEquals(7, app1.getClassTypes().size());
+        assertEquals(26, app1.getRelationTypes().size());
+
+        RelationType users = (RelationType) app1.getRelationTypes().stream().filter(r -> ((RelationType) r).getName().equals("users")).findFirst().orElseThrow();
+
+        ClassType userTransfer = (ClassType) app1.getClassTypes().stream().filter(c -> ((ClassType) c).getName().equals("RelationsTestModel::UserTransfer::ClassType")).findFirst().orElseThrow();
+        ClassType userView = (ClassType) app1.getClassTypes().stream().filter(c -> ((ClassType) c).getName().equals("RelationsTestModel::UserView::ClassType")).findFirst().orElseThrow();
+        ClassType mappedRelated = (ClassType) app1.getClassTypes().stream().filter(c -> ((ClassType) c).getName().equals("RelationsTestModel::MappedRelated::ClassType")).findFirst().orElseThrow();
+        ClassType mappedRelatedRow = (ClassType) app1.getClassTypes().stream().filter(c -> ((ClassType) c).getName().equals("RelationsTestModel::MappedRelatedRow::ClassType")).findFirst().orElseThrow();
+        ClassType unmappedRelated = (ClassType) app1.getClassTypes().stream().filter(c -> ((ClassType) c).getName().equals("RelationsTestModel::UnmappedRelated::ClassType")).findFirst().orElseThrow();
+        ClassType unmappedRelatedRow = (ClassType) app1.getClassTypes().stream().filter(c -> ((ClassType) c).getName().equals("RelationsTestModel::UnmappedRelatedRow::ClassType")).findFirst().orElseThrow();
+
+        assertEquals(userView, users.getTarget());
+
+        List<RelationType> userViewRelations = userView.getRelations();
+
+        assertEquals(25, userViewRelations.size());
+
+        // not sure why does JSLUtils produce aggregation == true for transients
```

Comment:
Because every transient reference is aggregated since payloads should always contain transient data (if there is)